In [52]:
import sys
sys.path.append('../')
import glob2 as glob
import json
import numpy as np
import pandas as pd

pd.options.display.max_columns = 200
pd.options.display.max_rows = 200
%matplotlib inline

import warnings; warnings.simplefilter('ignore')

In [53]:
pd.options.display.max_columns = 500
pd.options.display.max_rows = 500

In [54]:
# Importing Data
# path2data = '../Analysis/data/'

In [55]:
# Import Training and Test Data  
train_features = pd.read_csv('data2/dpc_train_features.csv', index_col='match_id')
train_targets = pd.read_csv('data2/dpc_train_target.csv', index_col='match_id')
test_features = pd.read_csv('data2/dpc_test_features.csv', index_col='match_id')
test_targets = pd.read_csv('data2/dpc_test_target.csv', index_col='match_id')

In [70]:
df1 = train_features
df2 = test_features

In [71]:
df1[['1is_pick', '2is_pick', '3is_pick', '4is_pick', '5is_pick', '6is_pick', '11is_pick', '12is_pick', '13is_pick', '14is_pick', '19is_pick', '20is_pick']] = 0
df2[['1is_pick', '2is_pick', '3is_pick', '4is_pick', '5is_pick', '6is_pick', '11is_pick', '12is_pick', '13is_pick', '14is_pick', '19is_pick', '20is_pick']] = 0

In [72]:
df1.head()

1is_pick  1team  1hero_id  2is_pick  2team  2hero_id  3is_pick  \
match_id                                                                     
5115031896         0      0        25         0      1        66         0   
4865407245         0      1        71         0      0        18         0   
4889341660         0      1       105         0      0       107         0   
5050335216         0      0        84         0      1        90         0   
4860032461         0      1       111         0      0        18         0   

            3team  3hero_id  4is_pick  4team  4hero_id  5is_pick  5team  \
match_id                                                                  
5115031896      0        60         0      1        90         0      0   
4865407245      1        55         0      0        66         0      1   
4889341660      1       121         0      0        80         0      1   
5050335216      0        19         0      1        79         0      0   
4860032461      1        69         0      0        55         0      1   

            5hero_id  6is_pick  6team  6hero_id  7is_pick  7team  7hero_id  \
match_id                                                                     
5115031896         7         0      1        96         1      0        85   
4865407245       129         0      0        96         1      1        37   
4889341660        86         0      0        66         1      1        57   
5050335216        60         0      1        23         1      0       103   
4860032461        65         0      0         8         1      1        66   

            8is_pick  8team  8hero_id  9is_pick  9team  9hero_id  10is_pick  \
match_id                                                                      
5115031896         1      1       102         1      1         6          1   
4865407245         1      0       121         1      0       103          1   
4889341660         1      0        18         1      0       111          1   
5050335216         1      1        96         1      1        86          1   
4860032461         1      0       121         1      0        16          1   

            10team  10hero_id  11is_pick  11team  11hero_id  12is_pick  \
match_id                                                                 
5115031896       0         52          0       0         13          0   
4865407245       1        100          0       1        106          0   
4889341660       1         95          0       1         96          0   
5050335216       0        106          0       0         61          0   
4860032461       1        107          0       1         42          0   

            12team  12hero_id  13is_pick  13team  13hero_id  14is_pick  \
match_id                                                                 
5115031896       1         19          0       0         46          0   
4865407245       0        104          0       1         16          0   
4889341660       0         11          0       1         55          0   
5050335216       1         10          0       0         93          0   
4860032461       0         46          0       1         10          0   

            14team  14hero_id  15is_pick  15team  15hero_id  16is_pick  \
match_id                                                                 
5115031896       1         98          1       1         86          1   
4865407245       0         76          1       0         65          1   
4889341660       0         52          1       0         79          1   
5050335216       1         41          1       1         37          1   
4860032461       0         54          1       0         37          1   

            16team  16hero_id  17is_pick  17team  17hero_id  18is_pick  \
match_id                                                                 
5115031896       0         20          1       1         72          1   
4865407245       1          8          1       0         46

In [73]:
df2.head()

1is_pick  1team  1hero_id  2is_pick  2team  2hero_id  3is_pick  \
match_id                                                                     
4870161807         0      1        66         0      0        97         0   
5103391298         0      0        60         0      1       102         0   
5062411190         0      0         9         0      1        66         0   
4971396775         0      1        72         0      0        66         0   
5059615080         0      1        97         0      0        41         0   

            3team  3hero_id  4is_pick  4team  4hero_id  5is_pick  5team  \
match_id                                                                  
4870161807      1        91         0      0        71         0      1   
5103391298      0         7         0      1        41         0      0   
5062411190      0        52         0      1        60         0      0   
4971396775      1        52         0      0        58         0      1   
5059615080      1         9         0      0       102         0      1   

            5hero_id  6is_pick  6team  6hero_id  7is_pick  7team  7hero_id  \
match_id                                                                     
4870161807       111         0      0        55         1      1       121   
5103391298        90         0      1        66         1      0        25   
5062411190        97         0      1        23         1      0        25   
4971396775        73         0      0        68         1      1         9   
5059615080        66         0      0        85         1      1        23   

            8is_pick  8team  8hero_id  9is_pick  9team  9hero_id  10is_pick  \
match_id                                                                      
4870161807         1      0        18         1      0        37          1   
5103391298         1      1         9         1      1        19          1   
5062411190         1      1         5         1      1        41          1   
4971396775         1      0        84         1      0        79          1   
5059615080         1      0        57         1      0       121          1   

            10team  10hero_id  11is_pick  11team  11hero_id  12is_pick  \
match_id                                                                 
4870161807       1         65          0       1         57          0   
5103391298       0         23          0       0         15          0   
5062411190       0        120          0       0         74          0   
4971396775       1        103          0       1         25          0   
5059615080       1        104          0       1         98          0   

            12team  12hero_id  13is_pick  13team  13hero_id  14is_pick  \
match_id                                                                 
4870161807       0        106          0       1         76          0   
5103391298       1         54          0       0         72          0   
5062411190       1         72          0       0         86          0   
4971396775       0         93          0       1        106          0   
5059615080       0         43          0       1         76          0   

            14team  14hero_id  15is_pick  15team  15hero_id  16is_pick  \
match_id                                                                 
4870161807       0         93          1       0         88          1   
5103391298       1         46          1       1         31          1   
5062411190       1        113          1       1        101          1   
4971396775       0         10          1       0         29          1   
5059615080       0        101          1       0          5          1   

            16team  16hero_id  17is_pick  17team  17hero_id  18is_pick  \
match_id                                                                 
4870161807       1         10          1       0        104          1   
5103391298       0         85          1       1          8

In [74]:
df1.to_csv('dpc_train_features.csv')
df2.to_csv('dpc_test_features.csv')

In [75]:
data1 = pd.read_csv('dpc_train_features.csv', index_col='match_id')
data2 = pd.read_csv('dpc_test_features.csv', index_col='match_id')

In [77]:
data1.shape

(1045, 66)

In [4]:
all_features = pd.concat([train_features, test_features])

# Index to split the training and test data sets
idx_split = train_features.shape[0]

In [5]:
all_features.head()

1is_pick  1team  1hero_id  2is_pick  2team  2hero_id  3is_pick  \
match_id                                                                     
5115031896         1      0        25         1      1        66         1   
4865407245         1      1        71         1      0        18         1   
4889341660         1      1       105         1      0       107         1   
5050335216         1      0        84         1      1        90         1   
4860032461         1      1       111         1      0        18         1   

            3team  3hero_id  4is_pick  4team  4hero_id  5is_pick  5team  \
match_id                                                                  
5115031896      0        60         1      1        90         1      0   
4865407245      1        55         1      0        66         1      1   
4889341660      1       121         1      0        80         1      1   
5050335216      0        19         1      1        79         1      0   
4860032461      1        69         1      0        55         1      1   

            5hero_id  6is_pick  6team  6hero_id  7is_pick  7team  7hero_id  \
match_id                                                                     
5115031896         7         1      1        96         1      0        85   
4865407245       129         1      0        96         1      1        37   
4889341660        86         1      0        66         1      1        57   
5050335216        60         1      1        23         1      0       103   
4860032461        65         1      0         8         1      1        66   

            8is_pick  8team  8hero_id  9is_pick  9team  9hero_id  10is_pick  \
match_id                                                                      
5115031896         1      1       102         1      1         6          1   
4865407245         1      0       121         1      0       103          1   
4889341660         1      0        18         1      0       111          1   
5050335216         1      1        96         1      1        86          1   
4860032461         1      0       121         1      0        16          1   

            10team  10hero_id  11is_pick  11team  11hero_id  12is_pick  \
match_id                                                                 
5115031896       0         52          1       0         13          1   
4865407245       1        100          1       1        106          1   
4889341660       1         95          1       1         96          1   
5050335216       0        106          1       0         61          1   
4860032461       1        107          1       1         42          1   

            12team  12hero_id  13is_pick  13team  13hero_id  14is_pick  \
match_id                                                                 
5115031896       1         19          1       0         46          1   
4865407245       0        104          1       1         16          1   
4889341660       0         11          1       1         55          1   
5050335216       1         10          1       0         93          1   
4860032461       0         46          1       1         10          1   

            14team  14hero_id  15is_pick  15team  15hero_id  16is_pick  \
match_id                                                                 
5115031896       1         98          1       1         86          1   
4865407245       0         76          1       0         65          1   
4889341660       0         52          1       0         79          1   
5050335216       1         41          1       1         37          1   
4860032461       0         54          1       0         37          1   

            16team  16hero_id  17is_pick  17team  17hero_id  18is_pick  \
match_id                                                                 
5115031896       0         20          1       1         72          1   
4865407245       1          8          1       0         46

In [8]:
df = all_features.copy()

In [ ]:
col2 = [f'{i}hero_id' for i in range(1, 23)]
print(col2)

In [24]:
is_pick_col = list(f'{i}is_pick' for i in range(1, 23))
print(is_pick_col)

['1is_pick', '2is_pick', '3is_pick', '4is_pick', '5is_pick', '6is_pick', '7is_pick', '8is_pick', '9is_pick', '10is_pick', '11is_pick', '12is_pick', '13is_pick', '14is_pick', '15is_pick', '16is_pick', '17is_pick', '18is_pick', '19is_pick', '20is_pick', '21is_pick', '22is_pick']


In [26]:
# Ban columns: [i]is_pick columns to replace with 0 
# 1-6, 11-14, 19-20
ban_cols = ['1is_pick', '2is_pick', '3is_pick', '4is_pick', '5is_pick', '6is_pick', '11is_pick', '12is_pick', '13is_pick', '14is_pick', '19is_pick', '20is_pick']

In [62]:
ban_cols = {
    '1is_pick': 0, 
     '2is_pick': 0, 
     '3is_pick': 0, 
     '4is_pick': 0, 
     '5is_pick': 0, 
     '6is_pick': 0, 
     '11is_pick': 0, 
     '12is_pick': 0, 
     '13is_pick': 0, 
     '14is_pick': 0, 
     '19is_pick': 0, 
     '20is_pick': 0
}

In [67]:
df[['1is_pick', '2is_pick', '3is_pick', '4is_pick', '5is_pick', '6is_pick', '11is_pick', '12is_pick', '13is_pick', '14is_pick', '19is_pick', '20is_pick']] = 0

In [69]:
df.dtypes

1is_pick     int64
1team        int64
1hero_id     int64
2is_pick     int64
2team        int64
2hero_id     int64
3is_pick     int64
3team        int64
3hero_id     int64
4is_pick     int64
4team        int64
4hero_id     int64
5is_pick     int64
5team        int64
5hero_id     int64
6is_pick     int64
6team        int64
6hero_id     int64
7is_pick     int64
7team        int64
7hero_id     int64
8is_pick     int64
8team        int64
8hero_id     int64
9is_pick     int64
9team        int64
9hero_id     int64
10is_pick    int64
10team       int64
10hero_id    int64
11is_pick    int64
11team       int64
11hero_id    int64
12is_pick    int64
12team       int64
12hero_id    int64
13is_pick    int64
13team       int64
13hero_id    int64
14is_pick    int64
14team       int64
14hero_id    int64
15is_pick    int64
15team       int64
15hero_id    int64
16is_pick    int64
16team       int64
16hero_id    int64
17is_pick    int64
17team       int64
17hero_id    int64
18is_pick    int64
18team      

In [64]:
df[['1is_pick', '2is_pick', '3is_pick', '4is_pick', '5is_pick', '6is_pick', '11is_pick', '12is_pick', '13is_pick', '14is_pick', '19is_pick', '20is_pick']].replace(to_replace, inplace=True)

1is_pick     None
2is_pick     None
3is_pick     None
4is_pick     None
5is_pick     None
6is_pick     None
11is_pick    None
12is_pick    None
13is_pick    None
14is_pick    None
19is_pick    None
20is_pick    None
dtype: object

In [58]:
df.loc[df['1is_pick', '2is_pick', '3is_pick', '4is_pick', '5is_pick', '6is_pick', '11is_pick', '12is_pick', '13is_pick', '14is_pick', '19is_pick', '20is_pick']] = 0

KeyError: ('1is_pick', '2is_pick', '3is_pick', '4is_pick', '5is_pick', '6is_pick', '11is_pick', '12is_pick', '13is_pick', '14is_pick', '19is_pick', '20is_pick')

In [50]:
df1 = df[ban_cols].map({1:0})

AttributeError: 'DataFrame' object has no attribute 'map'

In [42]:
df.head()

1is_pick  1team  1hero_id  2is_pick  2team  2hero_id  3is_pick  \
match_id                                                                     
5115031896         1      0        25         1      1        66         1   
4865407245         1      1        71         1      0        18         1   
4889341660         1      1       105         1      0       107         1   
5050335216         1      0        84         1      1        90         1   
4860032461         1      1       111         1      0        18         1   

            3team  3hero_id  4is_pick  4team  4hero_id  5is_pick  5team  \
match_id                                                                  
5115031896      0        60         1      1        90         1      0   
4865407245      1        55         1      0        66         1      1   
4889341660      1       121         1      0        80         1      1   
5050335216      0        19         1      1        79         1      0   
4860032461      1        69         1      0        55         1      1   

            5hero_id  6is_pick  6team  6hero_id  7is_pick  7team  7hero_id  \
match_id                                                                     
5115031896         7         1      1        96         1      0        85   
4865407245       129         1      0        96         1      1        37   
4889341660        86         1      0        66         1      1        57   
5050335216        60         1      1        23         1      0       103   
4860032461        65         1      0         8         1      1        66   

            8is_pick  8team  8hero_id  9is_pick  9team  9hero_id  10is_pick  \
match_id                                                                      
5115031896         1      1       102         1      1         6          1   
4865407245         1      0       121         1      0       103          1   
4889341660         1      0        18         1      0       111          1   
5050335216         1      1        96         1      1        86          1   
4860032461         1      0       121         1      0        16          1   

            10team  10hero_id  11is_pick  11team  11hero_id  12is_pick  \
match_id                                                                 
5115031896       0         52          1       0         13          1   
4865407245       1        100          1       1        106          1   
4889341660       1         95          1       1         96          1   
5050335216       0        106          1       0         61          1   
4860032461       1        107          1       1         42          1   

            12team  12hero_id  13is_pick  13team  13hero_id  14is_pick  \
match_id                                                                 
5115031896       1         19          1       0         46          1   
4865407245       0        104          1       1         16          1   
4889341660       0         11          1       1         55          1   
5050335216       1         10          1       0         93          1   
4860032461       0         46          1       1         10          1   

            14team  14hero_id  15is_pick  15team  15hero_id  16is_pick  \
match_id                                                                 
5115031896       1         98          1       1         86          1   
4865407245       0         76          1       0         65          1   
4889341660       0         52          1       0         79          1   
5050335216       1         41          1       1         37          1   
4860032461       0         54          1       0         37          1   

            16team  16hero_id  17is_pick  17team  17hero_id  18is_pick  \
match_id                                                                 
5115031896       0         20          1       1         72          1   
4865407245       1          8          1       0         46

In [38]:
df.columns(df[ban_cols]).replace(1,0, inplace=True)

TypeError: 'Index' object is not callable

In [36]:
df.head()

1is_pick  1team  1hero_id  2is_pick  2team  2hero_id  3is_pick  \
match_id                                                                     
5115031896         1      0        25         1      1        66         1   
4865407245         1      1        71         1      0        18         1   
4889341660         1      1       105         1      0       107         1   
5050335216         1      0        84         1      1        90         1   
4860032461         1      1       111         1      0        18         1   

            3team  3hero_id  4is_pick  4team  4hero_id  5is_pick  5team  \
match_id                                                                  
5115031896      0        60         1      1        90         1      0   
4865407245      1        55         1      0        66         1      1   
4889341660      1       121         1      0        80         1      1   
5050335216      0        19         1      1        79         1      0   
4860032461      1        69         1      0        55         1      1   

            5hero_id  6is_pick  6team  6hero_id  7is_pick  7team  7hero_id  \
match_id                                                                     
5115031896         7         1      1        96         1      0        85   
4865407245       129         1      0        96         1      1        37   
4889341660        86         1      0        66         1      1        57   
5050335216        60         1      1        23         1      0       103   
4860032461        65         1      0         8         1      1        66   

            8is_pick  8team  8hero_id  9is_pick  9team  9hero_id  10is_pick  \
match_id                                                                      
5115031896         1      1       102         1      1         6          1   
4865407245         1      0       121         1      0       103          1   
4889341660         1      0        18         1      0       111          1   
5050335216         1      1        96         1      1        86          1   
4860032461         1      0       121         1      0        16          1   

            10team  10hero_id  11is_pick  11team  11hero_id  12is_pick  \
match_id                                                                 
5115031896       0         52          1       0         13          1   
4865407245       1        100          1       1        106          1   
4889341660       1         95          1       1         96          1   
5050335216       0        106          1       0         61          1   
4860032461       1        107          1       1         42          1   

            12team  12hero_id  13is_pick  13team  13hero_id  14is_pick  \
match_id                                                                 
5115031896       1         19          1       0         46          1   
4865407245       0        104          1       1         16          1   
4889341660       0         11          1       1         55          1   
5050335216       1         10          1       0         93          1   
4860032461       0         46          1       1         10          1   

            14team  14hero_id  15is_pick  15team  15hero_id  16is_pick  \
match_id                                                                 
5115031896       1         98          1       1         86          1   
4865407245       0         76          1       0         65          1   
4889341660       0         52          1       0         79          1   
5050335216       1         41          1       1         37          1   
4860032461       0         54          1       0         37          1   

            16team  16hero_id  17is_pick  17team  17hero_id  18is_pick  \
match_id                                                                 
5115031896       0         20          1       1         72          1   
4865407245       1          8          1       0         46

In [ ]:
['1is_pick', '2is_pick', '3is_pick', '4is_pick', '5is_pick', '6is_pick', '7is_pick', '8is_pick', '9is_pick', '10is_pick', '11is_pick', '12is_pick', '13is_pick', '14is_pick', '15is_pick', '16is_pick', '17is_pick', '18is_pick', '19is_pick', '20is_pick', '21is_pick', '22is_pick']

In [17]:
is_pick_col

['1hero_id',
 '2hero_id',
 '3hero_id',
 '4hero_id',
 '5hero_id',
 '6hero_id',
 '7hero_id',
 '8hero_id',
 '9hero_id',
 '10hero_id',
 '11hero_id',
 '12hero_id',
 '13hero_id',
 '14hero_id',
 '15hero_id',
 '16hero_id',
 '17hero_id',
 '18hero_id',
 '19hero_id',
 '20hero_id',
 '21hero_id',
 '22hero_id']

In [25]:
df[is_pick_col]

1is_pick  2is_pick  3is_pick  4is_pick  5is_pick  6is_pick  \
match_id                                                                 
5115031896         1         1         1         1         1         1   
4865407245         1         1         1         1         1         1   
4889341660         1         1         1         1         1         1   
5050335216         1         1         1         1         1         1   
4860032461         1         1         1         1         1         1   
...              ...       ...       ...       ...       ...       ...   
4896666985         1         1         1         1         1         1   
5058096783         1         1         1         1         1         1   
5060130195         1         1         1         1         1         1   
4898190830         1         1         1         1         1         1   
5055077558         1         1         1         1         1         1   

            7is_pick  8is_pick  9is_pick  10is_pick  11is_pick  12is_pick  \
match_id                                                                    
5115031896         1         1         1          1          1          1   
4865407245         1         1         1          1          1          1   
4889341660         1         1         1          1          1          1   
5050335216         1         1         1          1          1          1   
4860032461         1         1         1          1          1          1   
...              ...       ...       ...        ...        ...        ...   
4896666985         1         1         1          1          1          1   
5058096783         1         1         1          1          1          1   
5060130195         1         1         1          1          1          1   
4898190830         1         1         1          1          1          1   
5055077558         1         1         1          1          1          1   

            13is_pick  14is_pick  15is_pick  16is_pick  17is_pick  18is_pick  \
match_id                                                                       
5115031896          1          1          1          1          1          1   
4865407245          1          1          1          1          1          1   
4889341660          1          1          1          1          1          1   
5050335216          1          1          1          1          1          1   
4860032461          1          1          1          1          1          1   
...               ...        ...        ...        ...        ...        ...   
4896666985          1          1          1          1          1          1   
5058096783          1          1          1          1          1          1   
5060130195          1          1          1          1          1          1   
4898190830          1          1          1          1          1          1   
5055077558          1          1          1          1          1          1   

            19is_pick  20is_pick  21is_pick  22is_pick  
match_id                                                
5115031896          1          1          1          1  
4865407245          1          1          1          1  
4889341660          1          1          1          1  
5050335216          1          1          1          1  
4860032461          1          1          1          1  
...               ...        ...        ...        ...  
4896666985          1          1          1          1  
5058096783          1          1          1          1  
5060130195          1          1          1          1  
4898190830          1          1          1          1  
5055077558          1          1          1          1  

[1162 rows x 22 columns]

In [12]:
merge_features_df = pd.concat([train_features, test_features])
#merge_targets_df = pd.concat([train_targets, test_targets])

# Index to split the training and test data sets
idx_split = train_features.shape[0]

heroes_df = df_full_features[[f'{i}hero_id' for i in range(1, 23)]]

NameError: name 'df_full_features' is not defined

In [27]:
all_features = pd.concat([train_features, test_features])

# Index to split the training and test data sets
idx_split = train_features.shape[0]

Woops, guess I doubled up on radiant win columns

In [5]:
dpc.drop(columns='Unnamed: 0', inplace=True)

In [6]:
#dpc.isna().sum()
dpc.fillna(0)

match_id  duration  leagueid  match_seq_num  series_id  series_type  \
0     4833593109      4036     10979     4062972494     337111            1   
1     4833371463      2976     10979     4062797533     337111            1   
2     4833120790      2700     10979     4062637540     337111            1   
3     4835973421      3504     10979     4064617551     337330            1   
4     4835092211      2530     10979     4063983045     337299            1   
...          ...       ...       ...            ...        ...          ...   
1157  5114927085       887     11280     4291834666     386573            1   
1158  5114498359      1365     11280     4291475848     386464            1   
1159  5056346258       988     11280     4241637919     374084            1   
1160  5060704777      1137     11280     4245399407     375084            1   
1161  5055077558      1223     11280     4240590934     373748            1   

      patch  dire_team_id  radiant_team_id  radiant_win_x  radiant_win_y  \
0        41     7118032.0         111474.0           True           True   
1        41      111474.0        7118032.0           True           True   
2        41      111474.0        7118032.0          False          False   
3        41     6214973.0        7059982.0          False          False   
4        41     7137328.0        7118032.0           True           True   
...     ...           ...              ...            ...            ...   
1157     41     7409177.0        6209143.0           True           True   
1158     41     7121518.0        2108395.0           True           True   
1159     41     6214973.0        7424508.0          False          False   
1160     41     7441136.0             67.0           True           True   
1161     41     7404529.0        2108395.0           True           True   

      1is_pick  1team  1hero_id  2is_pick  2team  2hero_id  3is_pick  3team  \
0         True  False        38      True   True       111      True  False   
1         True  False       111      True   True        38      True  False   
2         True  False        37      True   True        38      True  False   
3         True  False        66      True   True        52      True  False   
4         True   True        38      True  False        65      True   True   
...        ...    ...       ...       ...    ...       ...       ...    ...   
1157      True  False        85      True   True         7      True  False   
1158      True  False        59      True   True        66      True  False   
1159      True   True        97      True  False        72      True   True   
1160      True  False       102      True   True        86      True  False   
1161      True  False        19      True   True        60      True  False   

      3hero_id  4is_pick  4team  4hero_id  5is_pick  5team  5hero_id  \
0           55      True   True       107      True  False        46   
1           37      True   True        55      True  False        88   
2          111      True   True        55      True  False        88   
3          120      True   True        88      True  False        55   
4           66      True  False       111      True   True        88   
...        ...       ...    ...       ...       ...    ...       ...   
1157        90      True   True        72      True  False         9   
1158        34      True   True         7      True  False        57   
1159        84      True  False        66      True   True        79   
1160       100      True   True        25      True  False        97   
1161         3      True   True        72      True  False        79   

      6is_pick  6team  6hero_id  7is_pick  7team  7hero_id  8is_pick  8team  \
0         True   True        66      True  False        18      True   True   
1         True   True        66      True  False        16      True   True   
2         True   True        66      True  False        16      True   True   
3         True  

In [7]:
dpc.head()

match_id  duration  leagueid  match_seq_num  series_id  series_type  \
0  4833593109      4036     10979     4062972494     337111            1   
1  4833371463      2976     10979     4062797533     337111            1   
2  4833120790      2700     10979     4062637540     337111            1   
3  4835973421      3504     10979     4064617551     337330            1   
4  4835092211      2530     10979     4063983045     337299            1   

   patch  dire_team_id  radiant_team_id  radiant_win_x  radiant_win_y  \
0     41     7118032.0         111474.0           True           True   
1     41      111474.0        7118032.0           True           True   
2     41      111474.0        7118032.0          False          False   
3     41     6214973.0        7059982.0          False          False   
4     41     7137328.0        7118032.0           True           True   

   1is_pick  1team  1hero_id  2is_pick  2team  2hero_id  3is_pick  3team  \
0      True  False        38      True   True       111      True  False   
1      True  False       111      True   True        38      True  False   
2      True  False        37      True   True        38      True  False   
3      True  False        66      True   True        52      True  False   
4      True   True        38      True  False        65      True   True   

   3hero_id  4is_pick  4team  4hero_id  5is_pick  5team  5hero_id  6is_pick  \
0        55      True   True       107      True  False        46      True   
1        37      True   True        55      True  False        88      True   
2       111      True   True        55      True  False        88      True   
3       120      True   True        88      True  False        55      True   
4        66      True  False       111      True   True        88      True   

   6team  6hero_id  7is_pick  7team  7hero_id  8is_pick  8team  8hero_id  \
0   True        66      True  False        18      True   True        16   
1   True        66      True  False        16      True   True       102   
2   True        66      True  False        16      True   True        18   
3   True        37      True  False        16      True   True       102   
4  False        46      True   True        37      True  False        68   

   9is_pick  9team  9hero_id  10is_pick  10team  10hero_id  11is_pick  11team  \
0      True   True        37       True   False         65       True   False   
1      True   True        10       True   False         46       True   False   
2      True   True       121       True   False         46       True   False   
3      True   True        86       True   False         53       True   False   
4      True  False        16       True    True         52       True    True   

   11hero_id  12is_pick  12team  12hero_id  13is_pick  13team  13hero_id  \
0         57       True    True         88       True   False         39   
1         65       True    True         68       True   False        107   
2         65       True    True        114       True   False         57   
3         18       True    True         89       True   False        106   
4         86       True   False          7       True    True        106   

   14is_pick  14team  14hero_id  15is_pick  15team  15hero_id  16is_pick  \
0       True    True        121       True    True         67       True   
1       True    True         18       True    True        119       True   
2       True    True         95       True    True        107       True   
3       True    True         47       True    True         99       True   
4       True   False         82       True   False         42       True   

   16team  16hero_id  17is_pick  17team  17hero_id  18is_pick  18team  \
0   False         68       True    True          7       True   False   
1   False          5       True    True         23       True   False   
2   False         86       True    True         17       True   False   
3   False         68       

## Getting into it 

Recall Labels: 

- 'Team' = 0 = Radiant; 1=Dire
- Picks: ([i]is_pick)
    - First Phase: (1-10)
        - 1-6 BANS
        - 7-10 PICKS 
    - Second Phase: (11-18)
        - 11-14 B
        - 15-18 P
    - Third Phase: (19-22)
        - 19-20 B
        - 21-22 P

Subset cleaned data to get  match_id + radiant_win + drafts

In [8]:
#picks_cols = [str(i) for i in range(5) + 'hero_id']
#picks_ = dpc.iloc[:,[1,2,10:]]

In [9]:
picks = dpc.iloc[:, np.r_[0:2,10:77]]
picks = picks.rename(columns={'radiant_win_y': 'radiant_win'})

In [10]:
### Probably not the best way to do this, but I already had this snipped saved from when I was cleaning the data, so let's run with it. 

df = picks[['match_id', 'radiant_win', 'duration', 
                '1is_pick', '1team', '1hero_id',
                '2is_pick', '2team', '2hero_id', 
                '3is_pick', '3team', '3hero_id', 
                '4is_pick', '4team', '4hero_id', 
                '5is_pick', '5team', '5hero_id', 
                '6is_pick', '6team', '6hero_id', 
                '7is_pick', '7team', '7hero_id', 
                '8is_pick', '8team', '8hero_id', 
                '9is_pick', '9team', '9hero_id', 
                '10is_pick', '10team', '10hero_id', 
                '11is_pick', '11team', '11hero_id', 
                '12is_pick', '12team', '12hero_id',
                '13is_pick', '13team', '13hero_id', 
                '14is_pick', '14team', '14hero_id', 
                '15is_pick', '15team', '15hero_id', 
                '16is_pick', '16team', '16hero_id', 
                '17is_pick', '17team', '17hero_id',
                '18is_pick', '18team', '18hero_id', 
                '19is_pick', '19team', '19hero_id',
                '20is_pick', '20team', '20hero_id',
                '21is_pick', '21team', '21hero_id', 
                '22is_pick', '22team', '22hero_id']]

In [11]:
df

match_id  radiant_win  duration  1is_pick  1team  1hero_id  2is_pick  \
0     4833593109         True      4036      True  False        38      True   
1     4833371463         True      2976      True  False       111      True   
2     4833120790        False      2700      True  False        37      True   
3     4835973421        False      3504      True  False        66      True   
4     4835092211         True      2530      True   True        38      True   
...          ...          ...       ...       ...    ...       ...       ...   
1157  5114927085         True       887      True  False        85      True   
1158  5114498359         True      1365      True  False        59      True   
1159  5056346258        False       988      True   True        97      True   
1160  5060704777         True      1137      True  False       102      True   
1161  5055077558         True      1223      True  False        19      True   

      2team  2hero_id  3is_pick  3team  3hero_id  4is_pick  4team  4hero_id  \
0      True       111      True  False        55      True   True       107   
1      True        38      True  False        37      True   True        55   
2      True        38      True  False       111      True   True        55   
3      True        52      True  False       120      True   True        88   
4     False        65      True   True        66      True  False       111   
...     ...       ...       ...    ...       ...       ...    ...       ...   
1157   True         7      True  False        90      True   True        72   
1158   True        66      True  False        34      True   True         7   
1159  False        72      True   True        84      True  False        66   
1160   True        86      True  False       100      True   True        25   
1161   True        60      True  False         3      True   True        72   

      5is_pick  5team  5hero_id  6is_pick  6team  6hero_id  7is_pick  7team  \
0         True  False        46      True   True        66      True  False   
1         True  False        88      True   True        66      True  False   
2         True  False        88      True   True        66      True  False   
3         True  False        55      True   True        37      True  False   
4         True   True        88      True  False        46      True   True   
...        ...    ...       ...       ...    ...       ...       ...    ...   
1157      True  False         9      True   True        97      True  False   
1158      True  False        57      True   True        76      True  False   
1159      True   True        79      True  False        60      True   True   
1160      True  False        97      True   True        41      True  False   
1161      True  False        79      True   True        84      True  False   

      7hero_id  8is_pick  8team  8hero_id  9is_pick  9team  9hero_id  \
0           18      True   True        16      True   True        37   
1           16      True   True       102      True   True        10   
2           16      True   True        18      True   True       121   
3           16      True   True       102      True   True        86   
4           37      True  False        68      True  False        16   
...        ...       ...    ...       ...       ...    ...       ...   
1157        66      True   True        81      True   True        25   
1158        97      True   True        90      True   True         8   
1159        23      True  False        85      True  False        25   
1160        66      True   True        90      True   True        57   
1161        90      True   True        23      True   True        25   

      10is_pick  10team  10hero_id  11is_pick  11team  11hero_id  12is_pick  \
0          True   False         65       True   False         57       True   
1          True   False         46       True   False         65       True   
2          True   False         46       True  

In [12]:
df= df.astype('int')
df

match_id  radiant_win  duration  1is_pick  1team  1hero_id  2is_pick  \
0     4833593109            1      4036         1      0        38         1   
1     4833371463            1      2976         1      0       111         1   
2     4833120790            0      2700         1      0        37         1   
3     4835973421            0      3504         1      0        66         1   
4     4835092211            1      2530         1      1        38         1   
...          ...          ...       ...       ...    ...       ...       ...   
1157  5114927085            1       887         1      0        85         1   
1158  5114498359            1      1365         1      0        59         1   
1159  5056346258            0       988         1      1        97         1   
1160  5060704777            1      1137         1      0       102         1   
1161  5055077558            1      1223         1      0        19         1   

      2team  2hero_id  3is_pick  3team  3hero_id  4is_pick  4team  4hero_id  \
0         1       111         1      0        55         1      1       107   
1         1        38         1      0        37         1      1        55   
2         1        38         1      0       111         1      1        55   
3         1        52         1      0       120         1      1        88   
4         0        65         1      1        66         1      0       111   
...     ...       ...       ...    ...       ...       ...    ...       ...   
1157      1         7         1      0        90         1      1        72   
1158      1        66         1      0        34         1      1         7   
1159      0        72         1      1        84         1      0        66   
1160      1        86         1      0       100         1      1        25   
1161      1        60         1      0         3         1      1        72   

      5is_pick  5team  5hero_id  6is_pick  6team  6hero_id  7is_pick  7team  \
0            1      0        46         1      1        66         1      0   
1            1      0        88         1      1        66         1      0   
2            1      0        88         1      1        66         1      0   
3            1      0        55         1      1        37         1      0   
4            1      1        88         1      0        46         1      1   
...        ...    ...       ...       ...    ...       ...       ...    ...   
1157         1      0         9         1      1        97         1      0   
1158         1      0        57         1      1        76         1      0   
1159         1      1        79         1      0        60         1      1   
1160         1      0        97         1      1        41         1      0   
1161         1      0        79         1      1        84         1      0   

      7hero_id  8is_pick  8team  8hero_id  9is_pick  9team  9hero_id  \
0           18         1      1        16         1      1        37   
1           16         1      1       102         1      1        10   
2           16         1      1        18         1      1       121   
3           16         1      1       102         1      1        86   
4           37         1      0        68         1      0        16   
...        ...       ...    ...       ...       ...    ...       ...   
1157        66         1      1        81         1      1        25   
1158        97         1      1        90         1      1         8   
1159        23         1      0        85         1      0        25   
1160        66         1      1        90         1      1        57   
1161        90         1      1        23         1      1        25   

      10is_pick  10team  10hero_id  11is_pick  11team  11hero_id  12is_pick  \
0             1       0         65          1       0         57          1   
1             1       0         46          1       0         65          1   
2             1       0         46          1  

In [13]:
# Cleaned data set with all values converted to integers.
df.to_csv('data/DPC_converted.csv')

In [125]:
df1 = pd.read_csv('data/DPC_converted.csv', index_col=False)

In [126]:
df1.drop(columns='Unnamed: 0', inplace=True)

In [101]:
df1.set_index('match_id', inplace=True)

In [104]:
df1.head()

radiant_win  duration  1is_pick  1team  1hero_id  2is_pick  2team  \
match_id                                                                        
4833593109            1      4036         1      0        38         1      1   
4833371463            1      2976         1      0       111         1      1   
4833120790            0      2700         1      0        37         1      1   
4835973421            0      3504         1      0        66         1      1   
4835092211            1      2530         1      1        38         1      0   

            2hero_id  3is_pick  3team  3hero_id  4is_pick  4team  4hero_id  \
match_id                                                                     
4833593109       111         1      0        55         1      1       107   
4833371463        38         1      0        37         1      1        55   
4833120790        38         1      0       111         1      1        55   
4835973421        52         1      0       120         1      1        88   
4835092211        65         1      1        66         1      0       111   

            5is_pick  5team  5hero_id  6is_pick  6team  6hero_id  7is_pick  \
match_id                                                                     
4833593109         1      0        46         1      1        66         1   
4833371463         1      0        88         1      1        66         1   
4833120790         1      0        88         1      1        66         1   
4835973421         1      0        55         1      1        37         1   
4835092211         1      1        88         1      0        46         1   

            7team  7hero_id  8is_pick  8team  8hero_id  9is_pick  9team  \
match_id                                                                  
4833593109      0        18         1      1        16         1      1   
4833371463      0        16         1      1       102         1      1   
4833120790      0        16         1      1        18         1      1   
4835973421      0        16         1      1       102         1      1   
4835092211      1        37         1      0        68         1      0   

            9hero_id  10is_pick  10team  10hero_id  11is_pick  11team  \
match_id                                                                
4833593109        37          1       0         65          1       0   
4833371463        10          1       0         46          1       0   
4833120790       121          1       0         46          1       0   
4835973421        86          1       0         53          1       0   
4835092211        16          1       1         52          1       1   

            11hero_id  12is_pick  12team  12hero_id  13is_pick  13team  \
match_id                                                                 
4833593109         57          1       1         88          1       0   
4833371463         65          1       1         68          1       0   
4833120790         65          1       1        114          1       0   
4835973421         18          1       1         89          1       0   
4835092211         86          1       0          7          1       1   

            13hero_id  14is_pick  14team  14hero_id  15is_pick  15team  \
match_id                                                                 
4833593109         39          1       1        121          1       1   
4833371463        107          1       1         18          1       1   
4833120790         57          1       1         95          1       1   
4835973421        106          1       1         47          1       1   
4835092211        106          1       0         82          1       0   

            15hero_id  16is_pick  16team  16hero_id  17is_pick  17team  \
match_id                                                                 
4833593109         67          1       0         68          1       1   
4833371463        119          1       0          5 

In [172]:
df1.to_csv('data/DPC_pre_split.csv', index=False)

## Setting up prediction problem 

- Target is --> radiant_win

Leading up to this, I dropped the duration columns, as that information would not be known during the drafting phase of a match, only after the match is completed. 

For now I'll just go with radiant_win as my label (should I maybe include duration as a label?), and the drafting columns as my features. 

Let's hot encode and train/test split the data. Since we ended up having to drop the online matches from the first minor and major events, because they were played on patch 7.21, we are left with 1162 matches from our original ~1500, it follows that we will have to do a smaller train test split. Therefore, going forward, we will go with a 90-10 split (originally I was hoping for an 80-20 split). 

In [173]:
from sklearn.preprocessing import OneHotEncoder
from sklearn.model_selection import GridSearchCV, train_test_split
from sklearn.model_selection import ShuffleSplit, KFold

In [15]:
data = pd.read_csv((path2data + 'DPC_pre_split.csv'), index_col='match_id')

In [6]:
data.head()

radiant_win  duration  1is_pick  1team  1hero_id  2is_pick  2team  \
match_id                                                                        
4833593109            1      4036         1      0        38         1      1   
4833371463            1      2976         1      0       111         1      1   
4833120790            0      2700         1      0        37         1      1   
4835973421            0      3504         1      0        66         1      1   
4835092211            1      2530         1      1        38         1      0   

            2hero_id  3is_pick  3team  3hero_id  4is_pick  4team  4hero_id  \
match_id                                                                     
4833593109       111         1      0        55         1      1       107   
4833371463        38         1      0        37         1      1        55   
4833120790        38         1      0       111         1      1        55   
4835973421        52         1      0       120         1      1        88   
4835092211        65         1      1        66         1      0       111   

            5is_pick  5team  5hero_id  6is_pick  6team  6hero_id  7is_pick  \
match_id                                                                     
4833593109         1      0        46         1      1        66         1   
4833371463         1      0        88         1      1        66         1   
4833120790         1      0        88         1      1        66         1   
4835973421         1      0        55         1      1        37         1   
4835092211         1      1        88         1      0        46         1   

            7team  7hero_id  8is_pick  8team  8hero_id  9is_pick  9team  \
match_id                                                                  
4833593109      0        18         1      1        16         1      1   
4833371463      0        16         1      1       102         1      1   
4833120790      0        16         1      1        18         1      1   
4835973421      0        16         1      1       102         1      1   
4835092211      1        37         1      0        68         1      0   

            9hero_id  10is_pick  10team  10hero_id  11is_pick  11team  \
match_id                                                                
4833593109        37          1       0         65          1       0   
4833371463        10          1       0         46          1       0   
4833120790       121          1       0         46          1       0   
4835973421        86          1       0         53          1       0   
4835092211        16          1       1         52          1       1   

            11hero_id  12is_pick  12team  12hero_id  13is_pick  13team  \
match_id                                                                 
4833593109         57          1       1         88          1       0   
4833371463         65          1       1         68          1       0   
4833120790         65          1       1        114          1       0   
4835973421         18          1       1         89          1       0   
4835092211         86          1       0          7          1       1   

            13hero_id  14is_pick  14team  14hero_id  15is_pick  15team  \
match_id                                                                 
4833593109         39          1       1        121          1       1   
4833371463        107          1       1         18          1       1   
4833120790         57          1       1         95          1       1   
4835973421        106          1       1         47          1       1   
4835092211        106          1       0         82          1       0   

            15hero_id  16is_pick  16team  16hero_id  17is_pick  17team  \
match_id                                                                 
4833593109         67          1       0         68          1       1   
4833371463        119          1       0          5 

In [160]:
#data.drop(columns='Unnamed: 0', inplace=True)

In [7]:
data.shape

(1162, 68)

In [179]:
def hotencode(data):
    unq, idx = np.unique(data, return_inverse=1)
    col_idx = idx.reshape(data.shape)
    out = np.zeros((len(col_idx),col_idx.max()+1),dtype=int)
    out[np.arange(len(col_idx))[:,None], col_idx] = 1
    return pd.DataFrame(out, columns=unq, index=data.index)

Fuck I'm stupid, we're predicting a classification questions (win/notwin) --> can't have both 'radiant_win' and 'duration' as targets (I think).
But just incase duration might matter, or we want to explore it later, keep it in and simply drop it when we predict radiant_win. 

Targets = (radiant_win, duration); Features = (everthing else)

In [8]:
data

radiant_win  duration  1is_pick  1team  1hero_id  2is_pick  2team  \
match_id                                                                        
4833593109            1      4036         1      0        38         1      1   
4833371463            1      2976         1      0       111         1      1   
4833120790            0      2700         1      0        37         1      1   
4835973421            0      3504         1      0        66         1      1   
4835092211            1      2530         1      1        38         1      0   
...                 ...       ...       ...    ...       ...       ...    ...   
5114927085            1       887         1      0        85         1      1   
5114498359            1      1365         1      0        59         1      1   
5056346258            0       988         1      1        97         1      0   
5060704777            1      1137         1      0       102         1      1   
5055077558            1      1223         1      0        19         1      1   

            2hero_id  3is_pick  3team  3hero_id  4is_pick  4team  4hero_id  \
match_id                                                                     
4833593109       111         1      0        55         1      1       107   
4833371463        38         1      0        37         1      1        55   
4833120790        38         1      0       111         1      1        55   
4835973421        52         1      0       120         1      1        88   
4835092211        65         1      1        66         1      0       111   
...              ...       ...    ...       ...       ...    ...       ...   
5114927085         7         1      0        90         1      1        72   
5114498359        66         1      0        34         1      1         7   
5056346258        72         1      1        84         1      0        66   
5060704777        86         1      0       100         1      1        25   
5055077558        60         1      0         3         1      1        72   

            5is_pick  5team  5hero_id  6is_pick  6team  6hero_id  7is_pick  \
match_id                                                                     
4833593109         1      0        46         1      1        66         1   
4833371463         1      0        88         1      1        66         1   
4833120790         1      0        88         1      1        66         1   
4835973421         1      0        55         1      1        37         1   
4835092211         1      1        88         1      0        46         1   
...              ...    ...       ...       ...    ...       ...       ...   
5114927085         1      0         9         1      1        97         1   
5114498359         1      0        57         1      1        76         1   
5056346258         1      1        79         1      0        60         1   
5060704777         1      0        97         1      1        41         1   
5055077558         1      0        79         1      1        84         1   

            7team  7hero_id  8is_pick  8team  8hero_id  9is_pick  9team  \
match_id                                                                  
4833593109      0        18         1      1        16         1      1   
4833371463      0        16         1      1       102         1      1   
4833120790      0        16         1      1        18         1      1   
4835973421      0        16         1      1       102         1      1   
4835092211      1        37         1      0        68         1      0   
...           ...       ...       ...    ...       ...       ...    ...   
5114927085      0        66         1      1        81         1      1   
5114498359      0        97         1      1        90         1      1   
5056346258      1        23         1      0        85         1      0   
5060704777      0        66         1      1        90         1      1   
5055077558      0        90 

In [16]:
y = data[['radiant_win', 'duration']]
X = data.drop(columns=['radiant_win', 'duration'])

In [17]:
y.shape, X.shape

((1162, 2), (1162, 66))

In [18]:
from sklearn.preprocessing import OneHotEncoder
enc = OneHotEncoder(handle_unknown='ignore')
enc.fit(X)
OneHotEncoder(handle_unknown='ignore')

OneHotEncoder(categorical_features=None, categories=None, drop=None,
              dtype=<class 'numpy.float64'>, handle_unknown='ignore',
              n_values=None, sparse=True)

In [19]:
enc.categories_

[array([1]),
 array([0, 1]),
 array([  2,   3,   4,   6,   7,   9,  10,  16,  18,  19,  23,  25,  27,
         29,  31,  33,  37,  38,  40,  41,  42,  43,  46,  52,  53,  55,
         57,  58,  59,  60,  61,  62,  65,  66,  68,  69,  71,  72,  73,
         74,  75,  76,  79,  80,  81,  82,  83,  84,  85,  86,  87,  88,
         89,  90,  91,  93,  96,  97,  98,  99, 102, 103, 104, 105, 106,
        107, 111, 112, 119, 120, 121, 129]),
 array([1]),
 array([0, 1]),
 array([  2,   3,   4,   7,   9,  10,  16,  17,  18,  19,  22,  23,  25,
         29,  31,  33,  37,  38,  41,  42,  43,  46,  50,  52,  53,  54,
         55,  57,  58,  59,  60,  61,  65,  66,  67,  68,  71,  72,  73,
         74,  76,  78,  79,  80,  81,  83,  84,  85,  86,  88,  89,  90,
         91,  94,  96,  97,  98,  99, 102, 103, 105, 106, 107, 111, 112,
        119, 120, 121, 129]),
 array([1]),
 array([0, 1]),
 array([  2,   3,   4,   5,   6,   7,   9,  10,  14,  15,  16,  17,  18,
         19,  21,  23,  25,  29,  3

Splitting data (90% train, 10% test, random_state=17) and saving test target and features dataframes to csv

Split #2 csv's: random state 17, saved to data2 folder

Split #3 csv's: random state 17, hot encode before split, saved to data3 folder 

In [11]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.1, random_state = 17)

In [12]:
X_train.shape, X_test.shape, y_train.shape, y_test.shape

((1045, 66), (117, 66), (1045, 2), (117, 2))

## MADE SECOND SET OF TRAIN TEST AS PRECAUTION

- Also cause I forgot to save train data from first split
- Not 100% on whether i need to do encoding before or after split
    - but based on kaggle examples having data pre-split, and then encoding both train/test, I think it's fine to do after the split

In [13]:
np.bincount(y['radiant_win'])

array([596, 566])

In [14]:
## DONE @ 3am Sat march 22
X_train.to_csv('data2/dpc_train_features.csv', index='match_id')
y_train.to_csv('data2/dpc_train_target.csv', index='match_id')
X_test.to_csv('data2/dpc_test_features.csv', index='match_id')
y_test.to_csv('data2/dpc_test_target.csv', index='match_id')


### DONT TOUCH THESE NOW!!!!!!!!!!!!!

Test features and targets saved to csv @ approx 1:45 am 


In [187]:
X_test.to_csv('data/dpc_test_features.csv', index='match_id')
y_test.to_csv('data/dpc_test_target.csv', index='match_id')

In [191]:
X_test.shape, y_test.shape

((117, 66), (117, 2))

### Train/Validation Split 

Now, we turn our attention to the training set. 

Splitting training data into training and validation sets. Set validation set size to be 0.1, random state as 42 

In [192]:
from sklearn.model_selection import train_test_split
X_train, X_valid, y_train, y_valid = train_test_split(X_train, y_train, test_size=0.1, random_state =42)

In [193]:
X_train.shape, X_valid.shape, y_train.shape, y_valid.shape

((940, 66), (105, 66), (940, 2), (105, 2))

In [194]:
X_train.to_csv('data/dpc_train_features.csv', index='match_id')
y_train.to_csv('data/dpc_train_target.csv', index='match_id')
X_valid.to_csv('data/dpc_valid_features.csv', index='match_id')
y_valid.to_csv('data/dpc_valid_target.csv', index='match_id')

## FOLLOW UP QUESTIONS/CONSIDERATIONS

- should i have done onehotencoding before or after?
- 